src/tensorflow_encrypted/layer/core.py

```diff
@@ -1,13 +1,20 @@
 import numpy as np
+import math
+from ..protocol.pond import Pond
 
+DEFAULT_PROTOCOL = Pond
+
+# TODO
+# split backward function in compute_gradient and compute_backpropagated_error ?
 
 class Layer:
-    pass
+    prot = DEFAULT_PROTOCOL
```

Comment:
how about a `set_protocol` and `get_protocol` methods instead that update and read a global variable? was thinking something along the `with Pond(..) as prot:` pattern, where the `__enter__` method of a protocol calls `set_protocol`.

Reply:
okay, set_protocol and get_protocol functions of which class then?

Reply:
done